src/http/mod.rs

```diff
@@ -170,6 +199,18 @@ impl Request {
         self.headers.insert(req_header.0, req_header.1);
         self
     }
+
+    // Arg copied over as new header hashmap
+    pub fn with_params(mut self, req_params: HashMap<String, String>) -> Request {
+        self.params = req_params;
+        self
+    }
+
+    // Adds to existing hash map
+    pub fn with_param(mut self, req_param: (String, String)) -> Request {
```

Comment:
can we pass in 2 separate arguments here rather than a tuple? like with_param(mut self, key: String, value: String)